### PR TITLE
Add sanity checks for SwitchExpr inputs

### PR DESCRIPTION
**Pull request by mbasmanova**
Add checks to verify that all 'then' clauses and an optional 'else' clause of a
SWITCH expression have the same type.

### CODE REVIEW

File: velox/expression/SwitchExpr.cpp [hunk 1]
```diff
@@ -39,9 +40,38 @@ SwitchExpr::SwitchExpr(
...
+          thenType->toString(),
+          thenClause->type()->toString());
+    }
+  }
+
+  if (hasElseClause_) {
+    auto& elseClause = inputs_.back();
+    if (elseClause->type()->isUnKnown()) {
+      // Allow null constants.
+      VELOX_CHECK_NOT_NULL(dynamic_cast<ConstantExpr*>(elseClause.get()));
```
Comment (kgpai) [inline velox/expression/SwitchExpr.cpp:66]: nit: Curious, when do we have null constants for else clause ?

**Review by kgpai — APPROVED** [review]
Thanks for adding this

File: velox/expression/SwitchExpr.cpp [hunk 2]
```diff
@@ -39,9 +40,38 @@ SwitchExpr::SwitchExpr(
...
 
+  const auto& thenType = inputs_[1]->type();
+
   for (auto i = 0; i < numCases_; i++) {
     auto& condition = inputs_[i * 2];
     VELOX_CHECK_EQ(condition->type()->kind(), TypeKind::BOOLEAN);
+    auto& thenClause = inputs_[i * 2 + 1];
+    if (thenClause->type()->isUnKnown()) {
+      // Allow null constants.
+      VELOX_CHECK_NOT_NULL(dynamic_cast<ConstantExpr*>(thenClause.get()));
```
Comment (kgpai) [inline velox/expression/SwitchExpr.cpp:51]: Curious: when do we have null constants for then clause ?

**Review by xiaoxmeng — APPROVED** [review]
Nice change. Thanks!

File: velox/expression/SwitchExpr.cpp (hunk 2 above)
Comment (mbasmanova) [reply velox/expression/SwitchExpr.cpp:51]: For example, if (a > 5, null, 10).

**Pull request closed without merging**